src/js/components/agencyV2/visualizations/totalObligationsOverTime/AgencyBudgetLine.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+    data: PropTypes.arrayOf(PropTypes.shape({
+        period: PropTypes.number,
+        obligated: PropTypes.number
+    })),
+    xScale: PropTypes.func,
+    xDomain: PropTypes.array,
+    yScale: PropTypes.func,
+    height: PropTypes.number,
+    agencyBudget: PropTypes.number,
+    todaysDate: PropTypes.number,
+    padding: PropTypes.object
+};
+
+const AgencyBudgetLine = ({
+    data,
+    xScale,
+    xDomain,
+    yScale,
+    height,
+    agencyBudget,
+    todaysDate,
+    padding
+}) => {
+    const [show, setShow] = useState(false);
+
+    useEffect(() => {
+        if ((todaysDate >= xDomain[0]) && (todaysDate <= xDomain[1])) {
+            setShow(true);
+        }
+        else {
+            setShow(false);
+        }
+    }, [xScale, xDomain, todaysDate]);
+
+    return (
+        <g>
+            {show && <line
+                tabIndex="0"
+                className="total-budget-line"
+                x1={padding.left}
+                x2={xScale(todaysDate) + padding.left}
+                y1={height - yScale(agencyBudget - padding.top)}
+                y2={height - yScale(agencyBudget - padding.top)} />}
+            {show && <rect className="total-budget-difference" x={padding.left} y={height - yScale(agencyBudget - padding.top)} width={xScale(todaysDate)} height={yScale(agencyBudget) - yScale(data[data.length - 1].obligated) - padding.bottom} />}
```

Comment:
For the `y` property of the rectangle, we want scale the `agencyBudget`. Our `yScale`'s domain is based off of monetary values from the api, and its range is based off our true graphing area with the padding removed.  We want to pass monetary values to the `yScale` which will return the y-coordinate. We remove the padding after we scale the monetary value.
 
`y={height - yScale(agencyBudget) - padding.top}`

Reply:
We want to do the same thing for the `y1` and `y2` properties of the `line`

```
y1={height - yScale(agencyBudget) - padding.top}
y2={height - yScale(agencyBudget) - padding.top}
```